e2e/framework/utils/k8s_helper.go

```diff
@@ -251,3 +251,60 @@ func (k8sh *K8sHelper) IsThirdPartyResourcePresent(tprname string) bool {
 
 	return false
 }
+
+func (k8sh *K8sHelper) GetPodDetails(podNamePattern string, namespace string) (string, error) {
+	cmdArgs := []string{"get", "pods", "-l", "app=" + podNamePattern, "-o", "wide"}
+	if namespace != "" {
+		cmdArgs = append(cmdArgs, []string{"-n", namespace}...)
+	}
+	sout, serr, status := ExecuteCmd("kubectl", cmdArgs)
+	if status != 0 {
+		return serr, fmt.Errorf("Cannot find service")
+	}
+	return sout, nil
+}
+
+func (k8sh *K8sHelper) GetPodHostId(podNamePattern string, namespace string) (string, error) {
+	data, err1 := k8sh.GetPodDetails(podNamePattern, namespace)
+	if err1 != nil {
+		return data, err1
+	}
+
+	// Handle case when no data is returned
+	lines := strings.Split(data, "\n")
+	if len(lines) <= 1 {
+		return "NOT FOUND", fmt.Errorf("Cannot find Pod with Name matching %s", podNamePattern)
+	}
+
+	// Handle case when no data is returned
+	lines = lines[1 : len(lines)-1]
```

Comment:
looks like we're ignoring the 1st line, probably because that's the column headers? another way to do this would be to specify `--no-headers=true`.